### PR TITLE
Pantheon Deployment Workflow Update

### PR DESCRIPTION
**Pull request by srtfisher**
## Changes

- Remove `env:deploy` and all Terminus references from the multidev action. With the changes in deployment notifications, this step is no longer needed for multidev. There is still the argument of needing the cache-clearing call.
- Refactor the production deployment action to use a single job to inherit the commit message from the already checked-out code. This fixes the problem outlined below.
- Add timeouts to GitHub actions.

## Problem: Commits with quotes broke the deployment job

A commit with quotes broke the Terminus `env:deploy` command. For example, this commit message caused the command to break:

```
EXAMPLE-123: A merge that did something.

* Revert "Condensing the webpack back to a single one"

This reverts commit 48c0338.
```

Reading the commit message straight from Git itself can prevent this (https://l.alley.dev/880532d162 example failure for Alleyinz). 

Maybe there is a better way to do this that doesn't read from Git?

### CODE REVIEW

File: ci-templates/.github/workflows/deploy-to-pantheon-live.yml [hunk 1]
```diff
@@ -53,30 +53,22 @@ jobs:
...
-          ./terminus self:update
           sudo mv /tmp/terminus/terminus /usr/local/bin/terminus
 
       - name: Authenticate with Terminus
         run: terminus auth:login --machine-token=${{ secrets.PANTHEON_MACHINE_TOKEN }}
 
       - name: Deploy to Test
         run: |
-          terminus env:deploy ${{ env.PANTHEON_SITE }}.test --note="Deploying ${{ github.event.head_commit.message }} from GitHub Actions"
+          terminus env:deploy ${{ env.PANTHEON_SITE }}.test --note="$(git log -1 --pretty=%B)"
```
Comment (benpbolton) [inline ci-templates/.github/workflows/deploy-to-pantheon-live.yml]: what about 

```
commit_message="$( git log -1 --pretty=%B | sed 's/"/\\"/g' )"
terminus env:deploy ${{ env.PANTHEON_SITE }}.test --note="Deploying $commit_message from GitHub Actions"
```

Comment (benpbolton) [reply ci-templates/.github/workflows/deploy-to-pantheon-live.yml]: just be explicit about escaping the doublequotes? Still retain the 'fluff' around the commit message?

Comment (srtfisher) [reply ci-templates/.github/workflows/deploy-to-pantheon-live.yml]: That commit message could be multiple lines and wrapping it with quotes would look a little off. I think prefixing it with `Deploying from GitHub Actions:` could be a good compromise.

Comment (benpbolton) [reply ci-templates/.github/workflows/deploy-to-pantheon-live.yml]: The only thing we want to ensure is that a double quote in the git log won't break the wrapping of the terminus --note (so something like `commit_message="$( git log -1 --pretty=%B | sed 's/"/\\"/g' )"` might work?

**Review by GDKeller — APPROVED** [review]
㊙️

**Pull request merged**